src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,947 @@
+        telemetry.set_exception(exception=e, fault_type=consts.Diagnoser_Container_Check_Fault_Type, summary="Error occured while performing the diagnoser container checks")
+        diagnoser_output.append("An exception has occured while trying to perform diagnoser container check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+
+def executing_diagnoser_job(corev1_api_instance, batchv1_api_instance, filepath_with_timestamp, storage_space_available, absolute_path, helm_client_location, kubectl_client_location, release_namespace, security_policy_present):
+
+    global diagnoser_output
+    job_name = "azure-arc-diagnoser-job"
+    # CMD command to get helm values in azure arc and converting it to json format
+    command = [helm_client_location, "get", "values", "azure-arc", "--namespace", release_namespace, "-o", "json"]
+
+    # Using Popen to execute the helm get values command and fetching the output
+    response_helm_values_get = Popen(command, stdout=PIPE, stderr=PIPE)
+    output_helm_values_get, error_helm_get_values = response_helm_values_get.communicate()
+    if response_helm_values_get.returncode != 0:
+        if ('forbidden' in error_helm_get_values.decode("ascii") or 'timed out waiting for the condition' in error_helm_get_values.decode("ascii")):
+            telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                    summary='Error while doing helm get values azure-arc')
+
+    helm_values_json = json.loads(output_helm_values_get)
+
+    # Retrieving the proxy values if they are present
+    try:
+        is_proxy_enabled = helm_values_json["global"]["isProxyEnabled"]
+    except Exception as e:
+        is_proxy_enabled = False
+    try:
+        is_custom_cert = helm_values_json["global"]["isCustomCert"]
+    except Exception as e:
+        is_custom_cert = False
+    try:
+        proxy_cert = helm_values_json["global"]["proxyCert"]
+    except Exception as e:
+        proxy_cert = False
+
+    # Depending on the presence of proxy cert using the yaml
+    if proxy_cert and (is_custom_cert or is_proxy_enabled):
+        yaml_file_path = os.path.join(absolute_path, "troubleshoot_diagnoser_job_with_proxycert.yaml")
+    else:
+        yaml_file_path = os.path.join(absolute_path, "troubleshoot_diagnoser_job_without_proxycert.yaml")
+
+    # Setting the log output as Empty
+    diagnoser_container_log = ""
+    cmd_delete_job = [kubectl_client_location, "delete", "-f", ""]
+    cmd_delete_job[3] = str(yaml_file_path)
+
+    # Editing the yaml file based on the release nameespace
+    new_yaml = []
+    with open(yaml_file_path) as f:
+        list_doc = yaml.safe_load_all(f)
+        counter = 0
+        # USing release_namespace wherever required
```

Comment:
add the reason why we're deploying the job in release namespace in the comments. Mention that since we need access to the secrets in release namespace for performing helm get values we need this job to be running in the same namespace where release was installed